Fail
file path is :
projects/deltachat-core/rust/src/securejoin.rs

output is:


error is :
   Compiling deltachat v1.139.6 (/home/k/ogs_PHD/LLM4SE/2Rust_projects/auto_get_functionPair/projects/deltachat-core/rust)
error[E0425]: cannot find function `lookup_or_new` in this scope
  --> src/securejoin.rs:62:32
   |
62 |             let invitenumber = lookup_or_new(context, token::Namespace::InviteNumber, Some(chat_id)).await?;
   |                                ^^^^^^^^^^^^^ not found in this scope
   |
help: consider importing this function
   |
3  + use crate::token::lookup_or_new;
   |

error[E0425]: cannot find function `lookup_or_new` in this scope
  --> src/securejoin.rs:63:24
   |
63 |             let auth = lookup_or_new(context, token::Namespace::Auth, Some(chat_id)).await?;
   |                        ^^^^^^^^^^^^^ not found in this scope
   |
help: consider importing this function
   |
3  + use crate::token::lookup_or_new;
   |

error[E0425]: cannot find function `lookup_or_new` in this scope
  --> src/securejoin.rs:67:32
   |
67 |             let invitenumber = lookup_or_new(context, token::Namespace::InviteNumber, None).await?;
   |                                ^^^^^^^^^^^^^ not found in this scope
   |
help: consider importing this function
   |
3  + use crate::token::lookup_or_new;
   |

error[E0425]: cannot find function `lookup_or_new` in this scope
  --> src/securejoin.rs:68:24
   |
68 |             let auth = lookup_or_new(context, token::Namespace::Auth, None).await?;
   |                        ^^^^^^^^^^^^^ not found in this scope
   |
help: consider importing this function
   |
3  + use crate::token::lookup_or_new;
   |

warning: unused import: `crate::token::Namespace`
  --> src/securejoin.rs:35:5
   |
35 | use crate::token::Namespace;
   |     ^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> src/lib.rs:6:5
   |
6  |     unused,
   |     ^^^^^^
   = note: `#[warn(unused_imports)]` implied by `#[warn(unused)]`

error[E0277]: the trait bound `std::result::Result<_, anyhow::Error>: StdError` is not satisfied
  --> src/securejoin.rs:84:113
   |
84 |     let fingerprint = get_self_fingerprint(context).await.ok_or_else(|| bail!("Could not get self fingerprint"))?;
   |                                                                                                                 ^ the trait `StdError` is not implemented for `std::result::Result<_, anyhow::Error>`, which is required by `std::result::Result<std::string::String, anyhow::Error>: FromResidual<std::result::Result<Infallible, std::result::Result<_, anyhow::Error>>>`
   |
   = help: the following other types implement trait `FromResidual<R>`:
             <std::result::Result<T, F> as FromResidual<Yeet<E>>>
             <std::result::Result<T, F> as FromResidual<std::result::Result<Infallible, E>>>
   = note: required for `anyhow::Error` to implement `From<std::result::Result<_, anyhow::Error>>`
   = note: required for `std::result::Result<std::string::String, anyhow::Error>` to implement `FromResidual<std::result::Result<Infallible, std::result::Result<_, anyhow::Error>>>`

Some errors have detailed explanations: E0277, E0425.
For more information about an error, try `rustc --explain E0277`.
warning: `deltachat` (lib) generated 1 warning
error: could not compile `deltachat` (lib) due to 5 previous errors; 1 warning emitted
warning: build failed, waiting for other jobs to finish...
warning: `deltachat` (lib test) generated 1 warning (1 duplicate)
error: could not compile `deltachat` (lib test) due to 5 previous errors; 1 warning emitted
error: command `/home/k/.rustup/toolchains/1.77.1-x86_64-unknown-linux-gnu/bin/cargo test --no-run --message-format json-render-diagnostics` exited with code 101

